src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs

```diff
@@ -101,7 +101,7 @@ protected sealed override void InitializeWorker(AnalysisContext context)
             // However, this is a special analyzer that is directly invoked by the analysis host (IDE), so we do this check here.
             ReportDiagnostic severity;
             if (
-#if !CODE_STYLE
+#if !CODE_STYLE // Uses Roslyn 3.8 API
```

Comment:
@jmarolf When can we get the CodeStyle NuGet package inserted into the .NET SDK? That will avoid us from always lagging behind the Roslyn API and help clean this up further. Is it possible to get it into .NET5 RC1 (as off by default) or is it too late for it?

Reply:
> That will avoid us from always lagging behind the Roslyn API and help clean this up further

Unfortunately this is not correct. This repository frequently uses a newer version of Microsoft.Net.Compilers.Toolset than the minimum supported IDE version, and binding the code style layer to the toolset version would _completely_ break the dogfooding experience:

* Analyzers for this repository are configured to only show by default as live diagnostics in the IDE
* Analyzers bound to a newer version of the compiler than the IDE provides will fail to load for live analysis

Reply:
Hmm let me re-adjust my statement then. Inserting the CodeStyle NuGet package inserted into the .NET SDK will allow us to always reference the latest Roslyn compiler API and ensure that the analyzers layer never requires the `#if CODE_STYLE` stuff. For the code fixes layer, we have few choices:
1. Keep the current approach of referencing the last shipped Roslyn Workspace API, which mean code fixers will still need  `#if CODE_STYLE` for any new public APIs added to Workspaces that are not yet shipped.
2. Keep the current approach of referencing the last shipped Roslyn Workspace API, but any new code fixers that depend on unshipped public Workspace API will only be added to Features layer, not the shared project. Basically, the CodeStyle NuGet package gets new fixers only when it is able to reference the Workspaces API that those fixers require.
3. Remove code fixers completely from the CodeStyle layer and only ship them in the box with VS. This allows even the code fixer layer to _always_ reference the latest Workspaces API and we can remove all `#if CODE_STYLE` in entirety. This has the downside that someone referencing newer .NET SDK on lower version of VS will see some newer code style diagnostics but not fixes, unless they upgrade to newer VS. Another potential break is we change a shipped analyzer/code fix pair's handshake behavior (via properties/additional locations, etc.) then running a newer analyzer with an older code fix can lead to corrupt/crashing code fixes.

I feel approach 2 might be the best compromise, but we should discuss at design meeting. @sharwell can we bring this up at today's design meeting?

Reply:
> Inserting the CodeStyle NuGet package inserted into the .NET SDK will allow us to always reference the latest Roslyn compiler API ...

This is still not correct. It leads to exactly the situation described by https://github.com/dotnet/roslyn/pull/46873#discussion_r471558219.

Reply:
regardles @mavasani can you file a bug to get the code-style analyzers into the sdk? a pre-requisite for that is that we have all of them set to info or lower by default.

Reply:
> a pre-requisite for that is that we have all of them set to info or lower by default.

Ah good point. I think Formatting analyzer is warning by default.

Reply:
💭 I thought they were all hidden or suggestion by default.

Reply:
looks like all of them are except for IDE0043 (@mavasani is fixing this here: https://github.com/dotnet/roslyn/pull/46935)